model/validations.go

```diff
@@ -6,53 +6,57 @@ import (
 	"regexp"
 	"strconv"
 
+	"github.com/jinzhu/gorm"
 	"github.com/labstack/echo"
 )
 
 type Validations struct {
-	ID           int    `json:"questionID" db:"question_id"`
-	RegexPattern string `json:"regex_pattern" db:"regex_pattern"`
-	MinBound     string `json:"min_bound"  db:"min_bound"`
-	MaxBound     string `json:"max_bound"  db:"max_bound"`
+	ID           int    `json:"questionID"    db:"question_id"   gorm:"column:question_id"`
+	RegexPattern string `json:"regex_pattern" db:"regex_pattern" gorm:"column:regex_pattern"`
+	MinBound     string `json:"min_bound"     db:"min_bound"     gorm:"column:min_bound"`
+	MaxBound     string `json:"max_bound"     db:"max_bound"     gorm:"column:max_bound"`
 }
 
 func GetValidations(c echo.Context, questionID int) (Validations, error) {
 	validation := Validations{}
-	if err := db.Get(&validation, "SELECT * FROM validations WHERE question_id = ?",
-		questionID); err != nil {
+	if err := gormDB.Where("question_id = ?", questionID).First(&validation).Error; gorm.IsRecordNotFoundError(err) {
+		return Validations{}, nil
+	} else if err != nil {
 		c.Logger().Error(err)
 		return Validations{}, echo.NewHTTPError(http.StatusInternalServerError)
 	}
 	return validation, nil
 }
 
 func InsertValidations(c echo.Context, lastID int, validation Validations) error {
-	if _, err := db.Exec(
-		"INSERT INTO validations (question_id, regex_pattern, min_bound, max_bound) VALUES (?, ?, ?, ?)",
-		lastID, validation.RegexPattern, validation.MinBound, validation.MaxBound); err != nil {
+	validation.ID = lastID
+	if err := gormDB.Create(&validation).Error; err != nil {
 		c.Logger().Error(err)
 		return echo.NewHTTPError(http.StatusInternalServerError)
 	}
 	return nil
 }
 
 func UpdateValidations(c echo.Context, questionID int, validation Validations) error {
-	if _, err := db.Exec(
-		`INSERT INTO validations (question_id, regex_pattern, min_bound, max_bound) VALUES (?, ?, ?, ?)
-		ON DUPLICATE KEY UPDATE regex_pattern = ?, min_bound = ?, max_bound = ?`,
-		questionID,
-		validation.RegexPattern, validation.MinBound, validation.MaxBound,
-		validation.RegexPattern, validation.MinBound, validation.MaxBound); err != nil {
+	validationBefore := Validations{}
+
+	var err error
+	if validationBefore, err = GetValidations(c, questionID); gorm.IsRecordNotFoundError(err) {
+		return nil
+	} else if err != nil {
+		c.Logger().Error(err)
+		return echo.NewHTTPError(http.StatusBadRequest)
+	}
+
+	if err := gormDB.Model(&validationBefore).Update(&validation).Error; err != nil {
```

Comment:
```suggestion
	if err := gormDB.Model(&Validations{}).Update(&validation).Error; err != nil {
```
のようにすればUpdateの前にGetをしなくても大丈夫です。